src/mem/globalalloc.h

```diff
@@ -7,7 +7,7 @@
 namespace snmalloc
 {
   inline bool needs_initialisation(void*);
-  void* init_thread_allocator();
+  void* init_thread_allocator(std::function<void*(void*)>&);
```

Comment:
`init_thread_allocator` is hardcoded to use `current_alloc_pool`, so I don't think we should default `AllocPool` and `make_alloc_pool` to use these functions.

We probably want `AllocPool` and `make_alloc_pool` to be parametrized instead, and only specialize for the top-level `Alloc` and for `current_alloc_pool`.

Reply:
The refactor was much easier than I expected.